src/test/java/org/elasticsearch/search/suggest/ContextSuggestSearchTests.java

```diff
@@ -153,7 +155,26 @@ public void testMultiLevelGeo() throws Exception {
             assertEquals("Hotel Amsterdam in Berlin", suggestResponse.getSuggest().getSuggestion(suggestionName).iterator().next()
                     .getOptions().iterator().next().getText().string()); 
         }
-    }    
+    }
+
+    @Test
+    public void testMappingIdempotency() throws Exception {
+        XContentBuilder mapping = jsonBuilder().startObject().startObject(TYPE)
+                .startObject("properties").startObject("completion")
+                .field("type", "completion")
+                .startObject("context")
+                .startObject("location")
+                .field("type", "geo")
+                .array("precision", 1, 2, 3, 4)
+                .endObject()
+                .endObject().endObject()
+                .endObject().endObject();
+
+        assertAcked(prepareCreate(INDEX).setSettings(ImmutableSettings.builder().put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)).addMapping(TYPE, mapping.string()));
```

Comment:
The bug could be reproduced using a single shard, changed it to test with multiple shards